backend/src/services/enhanced_pdf_processor.py

```
@@ -100,7 +106,10 @@ async def process_file(self, request: ProcessingRequest) -> ProcessingResult:
             
             # Generate cache key based on file content hash and options
             file_path = Path(request.file_path)
-            file_hash = hashlib.md5(file_path.read_bytes()).hexdigest() if file_path.exists() else ""
+            file_hash = (
+                hashlib.md5(file_path.read_bytes()).hexdigest()
```

Ruff (0.13.1)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)